### DISCUSSION TITLE
[Arcropolis 3.7.2] CSS crash + mid-game freezing

### DISCUSSION THREAD

**Discussion by dre154** [Raytwo/ARCropolis#425] [Issues]
System Firmware Version
14.1.2

Atmosphere Version
1.3.2

Smash Version
13.0.1

ARCropolis Version
3.7.2

Can the crash be consistently reproduced?
Yes

At what moment does the crash happen?
The mod will cause crashes at random times when on the CSS menu, regardless if there is only 2 players or more. Crashes/freezing will also occur during gameplay OR when on the CSS

Details:
I had recently updated Arcropolis from version 3.5.1 to 3.6, then to 3.7.2. There were some issues with my mods in 3.6, however those got resolved....it was only after updating to 3.7.2 the game has henceforth spontaneously crashed with seemingly no trigger. The game itself will boot up fine and be able to make it past the main menu and into the character select screen...however after some time or a couple matches, the game will either abort entirely and crash or get "frozen" in place where the music is still playing and switch functions like volume and home button still work...but the game screen stays stuck with whatever was on it at the point of initial failure. The time before crashes is inconsistent, but eventually the crash WILL happen and usually ranges anywhere from 1 minute to 15+ minutes and will sometimes occur mid-match. I've tested everywhere from 2 players to 8 players and it seems to take longer to happen the less players there are...to add to this, I've also tested the game without ANY mods enabled and without any plugins BUT libarcropolis.nro and the crash still happens roughly around the 10 minute mark ( 3 times as of writing this), I've also tried playing the game without libarcropolis.nro enabled and it has crashed in the same way (2 times as of writing). I've ran my mods through Hekate's archive bit fix and all mods were "clean" but even so this crash still happens seemingly at random when at the CSS or playing the game.

Card has been formatted to FAT32, no corruption or other outside variables present on SD card.
Notably, most plugins besides the Ultimate training modpack plugins worked fine before 3.7.2, however since updating libarcropolis.nro seems to be the only stable thing as of writing.

Plugins turned on: Libarcropolis.nro
Plugins turned off: .DS_STORE, libarcropolis.nro, libbrawl_ness_effects.nro(Ness brawl effects plugin), libnn_hid_hook.nro, libnro_hook.nro, libparam_hook.nro, libsignpostgfx.nro (Sonic sign post victory screen), libsmashline_hook.nro, libone_slot_eff.nro (one-slot effects), libno_music_reset.nro (No music reset in training mode or otherwise), libtraining_modpack_menu.nro, libtraining_modpack.nro

Crash reports:
[01663136315_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9563489/01663136315_01006a800016e000.log)
[01662256919_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9563750/01662256919_01006a800016e000.log)
[01661914892_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9563751/01661914892_01006a800016e000.log)
![01661921436_01006a800016e000](https://user-images.githubusercontent.com/113347735/190088789-366d3f43-fe10-4f41-944c-12b15f79cbf3.jpg)
![01661914892_01006a800016e000](https://user-images.githubusercontent.com/113347735/190088809-3b9b0d72-9c3a-47ce-b3cd-5f30f05477b9.jpg)

Thanks! Appreciate all of your guys' work!

**Comment by dre154** [comment]
For what it's worth, there has been some speculation that the one-slot effects mod/plugin is the root of the issue and that it may be causing this "timed crash" in the CSS and during/after gameplay...though, as previously stated turning this AND almost all plugins off besides libarcropolis.nro off and running the game with no mod has still resulted in the same crash and conditions leading up to it, though it is worth noting: **after several attempts with booting up the game without any plugins besides libarcropolis.nro, the game seems to be stable with at least 4 players. More testing is needed though and I will update when I get the chance.**

**Reply by Raytwo** [reply]
From what I've been told since 3.5.0 or so, 4 players Smash is the max amount of players that can be loaded before things get funky, so that might have some impact too.

**Reply by dre154** [reply]
I kinda figured 8 players on a modded setup would get things messy for sure, so that can definitely be a key factor. I'm also hearing from others that smashline and the one-slot effect plugin may be another jumbled mess muddying the waters...I'll keep fiddling with plugins but so far those two are suspects.

Training modpack just straight up doesn't work anymore, though that's not much of a surprise.

**Comment by dre154** [comment]
UPDATE: So from various testing it would appear that something is bugging out with smashline...not sure what exactly WHY it's bugging out, but so far I've been able to test arcropolis pretty much with almost zero issues, as well as the one-slot effect being near perfectly stable minus like 1 or 2 random crashes....I'm currently testing out the training modpack with the two mentioned plugins turned on as well.

For anyone experiencing the same issue I'm having, go about the process by only running libarcropolis.nro  on for a couple of cycles (booting up the game and turning it off etc), then add one-slot effects if you have it. Keep smashline turned off for the time being; you're welcomed to test it yourself and you may have some success, but so far I haven't gotten a crash keeping it turned on. It might have something to do with data overflow or something....I'm not a dev so I have no clue, just making guesses but regardless, smashline SO FAR seems to be the issue.

**Comment by bandeyawx** [comment]
I (and multiple friends of mine who also use arcrop) found myself in a situation similar to yours, with same sys/arcrop/ams/smash version, but our crashing is strictly CSS. Mostly 3-4 player occurrences, but the occasional 2 player occurrence when there's a lot of character switching going on at once. 
From what I understand, the issue has something to do with limited system memory, but the issue is odd in that

-these crashes weren't occurring before the most recent arcrop releases
-there isn't really anything in the modpack I would consider memory expensive - things like the additional chara slots or stage alts, training modpack, sysdvr/dvr-patches, any plugins outside arcrop (outside one slot effects in a few tests but it didn't seem to affect the outcome) are all disabled and/or not installed
-my modpack's size, which I imagine would have something to do with it at some point, seemed largely irrelevant in previous releases where loading just as much content would have the game load completely stable with no issues regarding this amount of players for much longer play sessions.

I know this isn't specific to any one mod in ultimate/mods because I've gotten the game to crash with multiple different mod combinations - no skins and all stages, all stages no skins, no UI or added music, and what have you.
It feels like the "breaking point" to get this memory crash is coming much sooner than it used to, and I'm not really going out of my way to attempt to cause the crash in a lot of scenarios. Having one player switch to another character and flip through their alts will often be enough to crash.
Been troubleshooting all day with this one and none of your suggestions seem to have mitigated the issue, don't know how to go forward.

![01664496572_01006a800016e000](https://user-images.githubusercontent.com/101956709/193163190-a936669b-5768-492e-956c-4018220cae07.jpg)
[01664496572_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9679732/01664496572_01006a800016e000.log)
![01664494267_01006a800016e000](https://user-images.githubusercontent.com/101956709/193163191-7690c7d8-6256-4d08-9a7a-91fa5b698c53.jpg)
[01664494267_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/9679733/01664494267_01006a800016e000.log)
